config_builder.go

```diff
@@ -88,30 +106,56 @@ type ImplementationCfg struct {
 	// initializes all main lnd databases.
 	DatabaseBuilder func(ctx context.Context, logger btclog.Logger,
 		cfg *Config) (*DatabaseInstances, func(), error)
+
+	// ChainControlBuilder is a type that can provide a custom wallet
+	// implementation.
+	ChainControlBuilder
 }
 
 // DefaultWalletImpl is the default implementation of our normal, btcwallet
 // backed configuration.
 type DefaultWalletImpl struct {
+	cfg         *Config
+	logger      btclog.Logger
+	interceptor signal.Interceptor
+
+	pwService *walletunlocker.UnlockerService
+}
+
+// NewDefaultWalletImpl creates a new default wallet implementation.
+func NewDefaultWalletImpl(cfg *Config, logger btclog.Logger,
+	interceptor signal.Interceptor) *DefaultWalletImpl {
+
+	return &DefaultWalletImpl{
+		cfg:         cfg,
+		logger:      logger,
+		interceptor: interceptor,
+		pwService:   createWalletUnlockerService(cfg),
+	}
 }
 
 // RegisterRestSubserver is called after lnd creates the main proxy.ServeMux
 // instance. External subservers implementing this method can then register
 // their own REST proxy stubs to the main server instance.
 //
 // NOTE: This is part of the GrpcRegistrar interface.
-func (d *DefaultWalletImpl) RegisterRestSubserver(context.Context,
-	*proxy.ServeMux, string, []grpc.DialOption) error {
+func (d *DefaultWalletImpl) RegisterRestSubserver(ctx context.Context,
+	mux *proxy.ServeMux, restProxyDest string,
+	restDialOpts []grpc.DialOption) error {
 
-	return nil
+	return lnrpc.RegisterWalletUnlockerHandlerFromEndpoint(
+		ctx, mux, restProxyDest, restDialOpts,
+	)
 }
 
 // RegisterGrpcSubserver is called for each net.Listener on which lnd creates a
 // grpc.Server instance. External subservers implementing this method can then
 // register their own gRPC server structs to the main server instance.
 //
 // NOTE: This is part of the GrpcRegistrar interface.
-func (d *DefaultWalletImpl) RegisterGrpcSubserver(*grpc.Server) error {
+func (d *DefaultWalletImpl) RegisterGrpcSubserver(s *grpc.Server) error {
+	lnrpc.RegisterWalletUnlockerServer(s, d.pwService)
```

Comment:
Should `return` this

Reply:
That `RegisterWalletUnlockerServer` function doesn't return anything. The error in this `RegisterGrpcSubservers` method is just here for external subserver registration to report back any problems (such as LiT not being able to set up for whatever reason).